subsys/bluetooth/host/hci_core.c

```diff
@@ -403,13 +407,17 @@ static void rpa_timeout(struct k_work *work)
 	if (atomic_test_bit(bt_dev.flags, BT_DEV_ADVERTISING)) {
 		/* make sure new address is used */
 		set_advertise_enable(false);
-		le_set_private_addr(bt_dev.adv_id);
+		err |= le_set_private_addr(bt_dev.adv_id);
```

Comment:
why `|=` ? `err` is not changed before this line

Reply:
Just a habit, in case someone changes the order of the if statements. That said, the whole function could get a overhaul?

Reply:
Yeah, this may need some rethinkin

Reply:
Also, this kind of `|=` stuff doesn't really work if you want to retain the specific error number. Normally if some function returns an error you'd jump to some `failed:` label without trying to call any more functions that may fail.